### PR TITLE
base: authors REST API support

### PR DESCRIPTION
**Pull request by bouzlibop**
* Adds authors REST API support (addresses #217).

Signed-off-by: Adrian Pawel Baran <adrian.pawel.baran@cern.ch>

### CODE REVIEW

File: zenodo/base/format_templates/DataCite.xsl [hunk 1]
```diff
@@ -71,6 +71,20 @@ exclude-result-prefixes="marc fn dc invenio">
...
                     <xsl:value-of select="subfield[@code='a']"/>
                 </creatorName>
+                <xsl:for-each select="subfield[@code='0']">
+                    <xsl:if test="substring(., 2, 5) = 'orcid'">
+                        <nameIdentifier schemeURI="http://orcid.org" nameIdentifierScheme="ORCID">
+                            <!-- parse only id from (orcid)xxxx-xxxx-xxxx -->
+                            http://orcid.org/<xsl:value-of select="substring(., 8)"/>
+                        </nameIdentifier>
+                    </xsl:if>
+                    <xsl:if test="substring(., 2, 3) = 'gnd'">
```
Comment (lnielsen) [inline zenodo/base/format_templates/DataCite.xsl]: DataCite prevents having multiple. Will be fixed in future versions of the schema but we need to remove it for now.

File: zenodo/base/recordext/functions/resolve_authority.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def extract_authority_value(value, authority_control):
+    """Extract value of authority control.
+
+    :param value: single authority control value to extract
+    :param str authority_control: name of the authority control to be extracted
+    :return: extracted value or empty string
+    """
+    if (value or '').startswith('(' + authority_control + ')'):
```
Comment (lnielsen) [inline zenodo/base/recordext/functions/resolve_authority.py]: ``'(' + authority_control + ')'`` -> ``'(%s)' % authority_control``

File: zenodo/base/recordext/fields/zenodo.cfg [hunk 3]
```diff
@@ -134,7 +134,7 @@ authors:
         @only_if('_first_author' in self or '_additional_authors' in self)
         util_merge_fields_info_list(self, ['_first_author', '_additional_authors'])
     producer:
-        json_for_form(), {"creators": "[{'name': value.get('name',''), 'affiliation': value.get('affiliation',''), 'orcid': value.get('orcid', '')}]"}
+        json_for_form(), {"creators": "[{'name': value.get('name',''), 'affiliation': value.get('affiliation',''), 'gnd': value.get('gnd', ''), 'orcid': value.get('orcid', '')}]"}
         json_for_ld(), {"author": "[{'@type': 'Person', 'name': value.get('name',''), '@id': 'http://orcid.org/' + value.get('orcid','') if value.get('orcid') else None, }]"}
```
Comment (bouzlibop) [inline zenodo/base/recordext/fields/zenodo.cfg]: We don't need to update this one?

**Comment by lnielsen** [comment]
I think you like need to edit ``zenodo/modules/deposit/workflows/upload.py`` as well (``marshal_metadata_fields``)

**Comment by bouzlibop** [comment]
NOTE Test are probably failing because of not having https://github.com/inveniosoftware/invenio/pull/3254 integrated yet.

File: zenodo/base/format_templates/DataCite.xsl [hunk 4]
```diff
@@ -124,6 +140,24 @@ exclude-result-prefixes="marc fn dc invenio">
                         <nameIdentifier nameIdentifierScheme="info">info:eu-repo/grantAgreement/EC/FP7/<xsl:value-of select="subfield[@code='c']"/></nameIdentifier>
                     </contributor>
                 </xsl:for-each>
+                <xsl:for-each select="datafield[@tag=700][subfield[@code='4']='ths']">
```
Comment (bouzlibop) [inline zenodo/base/format_templates/DataCite.xsl]: Is it okay to  added `Supervisors` to `DataCite`? (It was in `DataCite3` but not `DataCite`)?

File: zenodo/base/templates/zenodo/api.html [hunk 5]
```diff
@@ -1980,6 +1982,8 @@ <h4 id="restapi-rep-files">Deposition file<small></small> <span class="pull-righ
 
 <h2 id="restapi-changes">Changes</h2>
 <dl>
+<dt>2015-10-06</dt>
```
Comment (bouzlibop) [inline zenodo/base/templates/zenodo/api.html]: what about date and message here?

**Pull request merged**